src/plugins/markdown-output/index.ts

```diff
@@ -0,0 +1,95 @@
+/**
+ * Plugin to output index.md files alongside index.html in the build directory.
+ * This copies the original markdown source files to their corresponding locations
+ * in the build output.
+ */
+
+import fs from 'node:fs/promises';
+import path from 'node:path';
+import { Plugin } from '@docusaurus/types';
+import logger from '@docusaurus/logger';
+
+/** Route prefixes to copy markdown files within. */
+const PREFIXES = ['docs/', 'blog/'];
+
+type Context = Parameters<Plugin['postBuild']>[0];
+
+async function fileExists(filePath: string): Promise<boolean> {
+  return fs.access(filePath).then(
+    () => true,
+    () => false,
+  );
+}
```

Comment:
Do permissions come into play here, or would it be simpler to just use `fs.existsSync`?

Reply:
I'm just curious because we generally use `fs.existsSync` as a replacement for `fs.pathExists` across ecosystem repos nowadays. I think having the `fs.existsSync` call is slightly more readable but also not a big deal either way.

Reply:
Was concerned that 370+ sync calls might be slower than async, but I measured it and it turns out to be faster! (_100ms sync / 160ms async_, n = 2.) Turns out the async overhead for these little fs calls is much higher than I expected!

Fixed in bc690b66bb1fb7392285767783c6fc78252b8aec and greatly simplifies the code. Good call! 😄 👍

Reply:
Woo, thanks for measuring!